extract/extract.js

```diff
@@ -0,0 +1,24 @@
+var numEvents = numEvents || 1;
+
+function skipNullOrEmptyValue(key, value) {
+    return (
+        value == null ||
+        (typeof value == 'object' && Object.keys(value).length == 0)
+    ) ? undefined : value;
+}
+
+// The original find query used the same regex on the object property.
+// However, many PR events don't have "problemroulette" in that property.
+
+db.event.find({
+    raw: {
+        $regex: /problemroulette/
+    }
+}).sort({
+    _id: -1
```

Comment:
I know nothing about bash scripts so I only tried review (or rather learn) readme documentation and extract.js. The js code is so concisely and beautifully written. As a newbie I did a lot of googling to fully understand the codes :D and learned a lot. I can hardly provide suggestions but I do have a few questions here. If anyone's free could you help me understand? Thanks!

About the objects {raw: { $regex: /problemroulette/ }} and  {_id: -1 } passed to find() and sort(),
1. I thought find() and sort() could only accept function objects as arguments, why is it feasible to pass these two objects even though they are not functions? I am guessing this involves some method overwriting..?
2. Where do they come from? Are they pre-defined in MongoDB? If so, is there any documentation which explains the parameters that can be referred to? 

Thanks in advance. 
Moreover, I feel I need to catch up with things about database and server, and maybe also shell.